presto-main/src/main/java/com/facebook/presto/type/BigintOperators.java

```diff
@@ -195,8 +195,10 @@ public static double castToDouble(@SqlType(StandardTypes.BIGINT) long value)
         return value;
     }
 
+    // x <= 20
     @ScalarOperator(CAST)
-    @SqlType(StandardTypes.VARCHAR)
+    @LiteralParameters("x")
+    @SqlType("varchar(x)")
```

Comment:
For all the cast operators we should check that the result doesn't exceed the result type length. And based on the cast type either to trim the result value to the result type, or throw the exception.